### DISCUSSION TITLE
"'Tensor' object is not callable"

### DISCUSSION THREAD

**Discussion by moushita-patnaik** [mrdbourke/pytorch-deep-learning#602] [Q&A]
Getting "Tensor object is not callable" error while running the training loop: 

```python
epochs = 1

for epoch in range(epochs): ## getting the error here 
  model_0.train() 

  y_pred = model_0(X_train)

  loss = loss_fn(y_pred, y_train)

  optimizer.zero_grad() 

  loss.backward()

  optimizer.step()

  model_0.eval() 
```

**Comment by mrdbourke** [comment]
Hi @Moushita ,

Can you print out `epoch` in the top of your for loop? 

I'm not sure what's happening here. 

What else have you tried so far? 

Is there any more code you can share?

**Comment by moushita-patnaik** [comment]
Hello. Thank you for responding!

I figured it out... Basically, i was declaring a variable called `range` which was overwriting the `range` method in the for loop. Hence the error. 😅
